optaplanner-core/src/main/java/org/optaplanner/core/impl/solver/DefaultSolver.java

```diff
@@ -183,12 +183,10 @@ public final void solve(Solution planningProblem) {
     public void outerSolvingStarted(DefaultSolverScope solverScope) {
         solving.set(true);
         basicPlumbingTermination.resetTerminateEarly();
-        solverScope.setStartingSystemTimeMillis(System.currentTimeMillis());
-        solverScope.setEndingSystemTimeMillis(null);
-        solverScope.setStartingSolverCount(0);
```

Comment:
startingSolverCount should only reset in the outerSolving, because it is used to determine if a solver is "started" or "restarted", see DefaultSolver#solvingStarted().